src/tournament_commands/player_commands/list.rs

```diff
@@ -6,5 +6,37 @@ use serenity::prelude::*;
 #[only_in(guild)]
 #[description("Lists out all tournament in the server.")]
 async fn list(ctx: &Context, msg: &Message, _: Args) -> CommandResult {
-    todo!()
+    /* Get references to needed data from context */
+    let data = ctx.data.read().await;
+    let name_and_id = data.get::<TournamentNameAndIDMapContainer>().unwrap();
+    let gld_tourns = data.get::<GuildAndTournamentIDMapContainer>().unwrap();
+
+    //Check if there are any active tournaments. If so list them, if not report that to the user
+    let id_iter = gld_tourns.get_left_iter(&msg.guild_id.unwrap()).unwrap();
+    let tourn_id = match id_iter.len() {
+        0 => {
+            msg.reply(
+                &ctx.http,
+                "There are no tournaments being held in this server.",
+            )
+            .await?;
+            return Ok(());
+        },
+        _ => {
+            let response = MessageBuilder::new();
+            for tourn in id_iter {
+                response
+                .push_bold_safe(tourn.tourn.name)
```

Comment:
This will not compile. `tourn` here is just a `TournamentId`, not a tournament. You'll need to use the `name_and_id` cycle map to get the tournament name from the id (alternatively, you could look up the tournament itself from a different map, but that would be generally less efficient.)

```suggestion
                .push_bold_safe(name_and_id.get_left(&tourn).unwrap())
```